docs/fsharp-cheatsheet.md

```diff
@@ -125,33 +138,56 @@ See [Literals (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/langua
 
 ## Functions
 
-The `let` keyword also defines named functions.
+Use the `let` keyword to define named functions.
 
     let negate x = x * -1
     let square x = x * x
-    let print x = printfn "The number is: %d" x
+    let print x = printfn $"The number is: {x}"
+
+### Pipe and Composition Operators
 
-### Pipe and composition operators
 Pipe operator `|>` is used to chain functions and arguments together. Double-backtick identifiers are handy to improve readability especially in unit testing:
 
     let ``square, negate, then print`` x =
         x |> square |> negate |> print
 
-This operator can assist the F# type checker by providing type information before use:
-
-    let sumOfLengths (xs : string []) =
-        xs
-        |> Array.map (fun s -> s.Length)
-        |> Array.sum
-
 Composition operator `>>` is used to compose functions:
 
-    let squareNegateThenPrint' =
+    let squareNegateThenPrint =
         square >> negate >> print
 
-<div id="recursive-functions"></div>
+### Signatures and Explicit Typing
+
+Function signatures are useful for quickly learning the input and output of functions. The last type is the return type and all preceding types are the input types.
+
+    int -> string                       // this defines a function that receives an integer; returns a string
+    int -> int -> string                // two integer inputs; returns a string
+    unit -> string                      // no inputs; returns a string
+    string -> unit                      // accepts a string; no return
+    (int * string) -> string -> string  // a tuple of int and string and a string inputs; returns a string
+
+Most of the time, the compiler can determine the type of a parameter, but there are cases may you wish to be explicit or the compiler needs a hand.
+Here is a function with a signature `string -> char -> int` and the input and return types are explicit:
 
-### Recursive Functions
+    let countWordsStartingWithLetter (theString: string) (theLetter: char) : int =
+        theString.Split ' '
+        |> Seq.where (fun (word:string) -> word.StartsWith theLetter)  // explicit typing in a lambda
+        |> Seq.length
+
+Examples of functions that take `unit` (i.e. nothing) as arguments and return different [Collection](#collections) types.
```

Comment:
again, it's not nothing - its the single value for the unit type, and I dont think its useful to cal it nothing

Reply:
Will resolve this one and use the one above as the reminder.